js/modules/k6/html/serialize.go

```diff
@@ -0,0 +1,95 @@
+/*
+ *
+ * k6 - a next-generation load testing tool
+ * Copyright (C) 2016 Load Impact
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package html
+
+import (
+	neturl "net/url"
+
+	"github.com/PuerkitoBio/goquery"
+	"github.com/dop251/goja"
+)
+
+type FormValue struct {
+	Name  string
+	Value goja.Value
+}
+
+// nolint: goconst
+func (s Selection) SerializeArray() []FormValue {
+	submittableSelector := "input,select,textarea,keygen"
+	var formElements *goquery.Selection
+	if s.sel.Is("form") {
+		formElements = s.sel.Find(submittableSelector)
+	} else {
+		formElements = s.sel.Filter(submittableSelector)
```

Comment:
Could we add a test case for this code path as well please.

Reply:
Will do